### PR TITLE
HAC 63: Split core API dependencies from console dependencies

### PR DESCRIPTION
**Pull request by florkbr**
This pull request is based off of @invincibleJai's dependency analysis here: https://docs.google.com/spreadsheets/d/19n-eJr-y6DMnYr0n0aDEfluKz4FC6XRw2Ko556wZeQk

This PR will setup separate files for distinguishing between the core SDK API dependencies versus what is required by the openshift console SDK flavor. While there should be no functional difference, the main benefit is setting us up for migrating to the new repository and pulling in the source for core SDK dependencies one by one.

Core SDK files:
`core-api.ts`
`internal-api.ts`

Openshift Console SDK files:
`core-api-console.ts`
`internal-api-console.ts`

When we cut over to the new repo, we will need to modify where we pull in the console SDK files from.

See: https://issues.redhat.com/browse/HAC-63

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/api/core-api-console.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable */
+import {
+  UseActivePerspective,
+} from '../extensions/console-types';
+
+export const useActivePerspective: UseActivePerspective = require('@console/dynamic-plugin-sdk/src/perspective/useActivePerspective')
+  .default;
```
Comment (andrewballantyne) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/core-api-console.ts:7]: Probably want EOF new lines.

**Review by andrewballantyne — COMMENTED** [review]
This looks good as a first step, we'll keep build on it with more PRs in the future. 

I think if we can keep them separated as we move towards leaving OpenShift Console for the other repo, it'll help structure things as to make the switchover faster later.

**Review by andrewballantyne — APPROVED** [review]
/assign @vojtechszocs 

You're okay with this structure as a means to start separating the code in prep for moving it over?

**Review by andrewballantyne — APPROVED** [review]
/lgtm
/retest

**Comment by vojtechszocs** [comment]
> You're okay with this structure as a means to start separating the code in prep for moving it over?

@andrewballantyne @florkbr Looks good, I'm just wondering about naming it `dynamic-core-api` - if I understand correctly, `core-api-console.ts` should be conceptually a superset of `core-api.ts` is that correct?

If so, we can update `packages/console-dynamic-plugin-sdk/src/lib-core.ts` to point to a new `core-api-console.ts` module that would re-export * from `core-api.ts` module.

**Comment by andrewballantyne** [comment]
> > You're okay with this structure as a means to start separating the code in prep for moving it over?
> 
> @andrewballantyne @florkbr Looks good, I'm just wondering about naming it `dynamic-core-api` - if I understand correctly, `core-api-console.ts` should be conceptually a superset of `core-api.ts` is that correct?
> 
> If so, we can update `packages/console-dynamic-plugin-sdk/src/lib-core.ts` to point to a new `core-api-console.ts` module that would re-export * from `core-api.ts` module.

For posterity, we discussed it [on slack](https://coreos.slack.com/archives/C02EQERSV3N/p1637342792356500?thread_ts=1637341744.354900&cid=C02EQERSV3N) and concluded this is fine for now; future PRs will continue to reorganize things.

**Comment by andrewballantyne** [comment]
@florkbr you're probably going to need QE to jump in here and do a test to get the qe-approved label.

/cc @yapei 

We are not documenting the SDK at this time, so adding the other labels.
/label docs-approved
/label px-approved

**Comment by yapei** [comment]
deployed `console-demo-plugin` plugin and did a regression of dynamic plugin cases, no issues found
/label qe-approved

**Pull request merged**